### ISSUE TITLE
Tracked exception has no related traces

### ISSUE THREAD

**Issue by scott-lin** [Microsoft/ApplicationInsights-node.js#386]
Hi,

I'm logging traces and then an exception to Application Insights, however, the Exception relates to zero traces in the portal. I'm using manual telemetry collection and `"applicationinsights": "^1.0.2"`.

I saw issue https://github.com/Microsoft/ApplicationInsights-node.js/issues/37, which looked related to my problem, but the solution did not work. Perhaps the library has changed since then.

I'm attempting to set `operationId` of the telemetry client hoping to relate traces and exceptions. Here's the gist of what I am doing:

```
import appInsights = require('applicationinsights');
import { v4 as uuid } from 'uuid';

let telemetryClient = new appInsights.TelemetryClient('some iKey');
telemetryClient.context.tags[telemetryClient.context.keys.operationId] = uuid();

telemetryClient.trackTrace({ message: 'some message' });
telemetryClient.trackException({ exception: someException });
```

What's the proper way to relate all telemetry tracked through a single client instance? Soon I will be tracking metrics through the same client and will want those to relate as well.

Portal screenshot for context:

![image](https://user-images.githubusercontent.com/31644762/38850802-e68f2e2c-41c7-11e8-9126-b41873340c1a.png)

**Comment by OsvaldoRosado** [comment]
Assuming that all expected telemetry is present if you click the blue "View in Analytics" icon to the right of the Operation ID, it looks like you're running into a UI-level issue.

What I expect is happening is that while you are setting `operation_Id` to put all telemetry in the same global context, you are not also setting `operation_parentId` to create parent-child relationships for events within the operation (eg. usually dependencies are children of requests). In the version of the UI you are viewing, this is important because traces only appear in the bottom panel, and the bottom panel can only be shown for items in the timeline, and their children. Thus, if the trace is not a child of a telemetry type permitted in the timeline (request, dependency, exception), that version of the UI can't display it.

This being said, it appears you're looking at an older version of the UI. Could you try clearing your cache and restarting your browser? The newer version doesn't have the described limitation in the bottom panel and should show you every telemetry item within the operation. You can tell if you're seeing the newer version because the bottom panel has this updated design:
<img width="955" alt="image" src="https://user-images.githubusercontent.com/4926351/38851185-8830519c-41c9-11e8-93c3-602b2995b35d.png">

Out of curiosity, in your scenario, what triggers new "operation" contexts to start? If this is a typical Node.js HTTP server scenario, setting of operation and parent ids should be done automatically in the default configuration upon each new incoming request and should not require (and will actually be disrupted by) overriding.

**Comment by scott-lin** [comment]
@OsvaldoRosado, thanks for the swift and helpful reply. Indeed, I seemed to have been using an old UI version and clearing the cache + refresh shows the same you have with my expected traces.

To answer your question, my application is not a web service actually. It's a [VSTS extension](https://docs.microsoft.com/en-us/vsts/extend/overview?view=vsts) that I would like to track telemetry for. I figured I would use Application Insights to track important metrics and errors, since we are not able to easily access and aggregate VSTS task logs.

My goal is to be able to track exceptions our application encounters in the wild, and if there are not enough details for debugging, view the traces leading up the the exception. To "group" traces and exceptions, I was hoping to use the `operation_Id` solution I mentioned above. If there is a better way of doing this outside of a web service (request) context, I'm all ears.
